sources/osg/Matrix.js

```diff
@@ -611,6 +649,22 @@ var Matrix = {
         };
     } )(),
 
+    preMultRotate: function ( matrix, quat ) {
+        if ( quat[ 0 ] === 0.0 && quat[ 1 ] === 0.0 && quat[ 2 ] === 0.0 && quat[ 3 ] === 1.0 )
+            return;
+        var r = Matrix.create();
+        Matrix.makeRotateFromQuat( r, quat );
+        Matrix.preMult( matrix, r );
+    },
+
+    postMultRotate: function ( matrix, quat ) {
+        if ( quat[ 0 ] === 0.0 && quat[ 1 ] === 0.0 && quat[ 2 ] === 0.0 && quat[ 3 ] === 1.0 )
+            return;
+        var r = Matrix.create();
```

Comment:
@stephomi About this, I don't see how to do Matrix.create() in a closure. None of the methods in matrix does that. All of them create the matrix as I did, inside the function body. Do you mean to have only the var name in the closure like this?  https://github.com/cedricpinson/osgjs/blob/develop/sources/osg/Matrix.js#L1095